docs/conf.py

```diff
@@ -285,6 +285,38 @@
 needs_default_layout = "clean"
 needs_default_style = None
 
+needs_grids = {
```

Comment:
these are really complex, stringly-typed dictionaries. Is it possible to replace any of the dicts used in this PR with classes?